dags/iop/parser.py

```diff
@@ -100,3 +105,80 @@ def _get_extracted_arxiv_eprint_value(self, article):
             self.logger.error("The arXiv value is not valid.", dois=self.dois)
         except AttributeError:
             self.logger.error("No arXiv eprints found", dois=self.dois)
+
+    def _get_authors(self, article):
+        contrib_types = article.findall(
+            "front/article-meta/contrib-group/contrib[@contrib-type='author']"
+        )
+        authors = []
+        surname = ""
+        given_names = ""
+
+        for contrib_type in contrib_types:
+            try:
+                surname = contrib_type.find("name/surname").text
+            except AttributeError:
+                self.logger.error("Surname is not found in XML", dois=self.dois)
+            try:
+                given_names = contrib_type.find("name/given-names").text
+            except AttributeError:
+                self.logger.error("Given_names is not found in XML", dois=self.dois)
+
+            reffered_ids = contrib_type.findall("xref[@ref-type='aff']")
+            affiliations = [
+                self._get_affiliation_value(article, reffered_id)
+                for reffered_id in reffered_ids
+                if self._get_affiliation_value(article, reffered_id) is not None
+            ]
+
+            if "collaboration" not in given_names.lower():
+                author = clean_dict(
+                    {
+                        "surname": surname,
+                        "given_names": given_names,
+                        "affiliations": affiliations,
+                    }
+                )
+                authors.append(author)
+
+        if all(authors):
+            return authors
+
+    def _get_affiliation_value(self, article, reffered_id):
+        institution_and_country = {}
+        try:
+            id = reffered_id.get("rid")
+        except AttributeError:
+            self.logger.error("Referred id is not found")
+        institution = self._get_institution(article, id)
+        country = self._get_country(article, id)
+        try:
+            institution_and_country = {"country": country}
+            institution_and_country.update(
+                {"institution": ", ".join([institution, country])}
+            )
+        except TypeError:
+            self.logger.error("Cannot join institution and country to one value")
+        cleaned_institution_and_country = clean_dict(institution_and_country)
```

Comment:
I don't like it tbh, as I mentioned I would prefer to check if the value was extracted and then add it to the dict instead of iterating through the whole dictionary to remove empty keys